### ISSUE TITLE
Enhance Filter options

### ISSUE THREAD

**Issue by iluvadev** [iluvadev/PocketBaseClient-csharp#4]

**Comment by iluvadev** [comment]
In PocketBaseClient version 0.3.0, the filtering options were 
* With specific functions for every field
* Uniques, for only one field, without AND/OR support
* Limited to PocketBase operators
* With the need to materialize the filter with `GetItems()`
```csharp
var myTodos = myTodosCollection.Filter(todo => todo.Name(OperatorText.Like, "my todo")).GetItems();
```

**Comment by iluvadev** [comment]
Added in PocketBaseClient v.0.4.0 a Sort system to sort Filtered data:
```csharp
var myTodos = myTodosCollection.Filter(todo => todo.Title.StartsWith("my todo"))
                               .SortBy(todo => todo.Created.Desc().AndThenBy(todo.Updated))
foreach(var myTodo in myTodos)
{
   ...
}
```

**Comment by iluvadev** [comment]
Issue solved using code generated with [pbcodegen v0.5.0](https://github.com/iluvadev/PocketBaseClient-csharp/releases/tag/pbcodegen_v0.5.0)
